etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+*
+* @name quotes
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [quotes]{@link https://eslint.org/docs/rules/quotes}
+*/
+rules[ 'quotes' ] = 'off';
+
+/**
+* Enforces passing the radix parameter to the `parseInt()` function.
+*
+* @name radix
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [radix]{@link https://eslint.org/docs/rules/radix}
+*
+* @example
+* // Bad...
+* const bad = parseInt('123');
+*
+* // Good...
+* const good = parseInt('123', 10);
+*/
+rules[ 'radix' ] = 'error';
+
+/**
+* Disables the requirement for `await` in `async` functions.
+*
+* @name require-await
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [require-await]{@link https://eslint.org/docs/rules/require-await}
+*/
+rules[ 'require-await' ] = 'off';
+
+/**
+* Disables the rule that requires or disallows semicolons instead of ASI.
+*
+* @name semi
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [semi]{@link https://eslint.org/docs/rules/semi}
+*/
+rules[ 'semi' ] = 'off';
+
+/**
+* Enforces consistent spacing before function parentheses.
+*
+* @name @typescript-eslint/space-before-function-paren
+* @memberof rules
+* @type {Array}
+* @default [ 'error', { 'anonymous': 'always', 'named': 'never', 'asyncArrow': 'always' }]
+* @see [space-before-function-paren]{@link https://typescript-eslint.io/rules/space-before-function-paren}
+*/
+rules[ '@typescript-eslint/space-before-function-paren' ] = [
+	'error',
+	{
+		'anonymous': 'always',
+		'named': 'never',
+		'asyncArrow': 'always'
+	}
+];
+
+/**
+* Enforces consistent spacing inside parentheses.
+*
+* @name space-in-parens
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'always' ]
+* @see [space-in-parens]{@link https://eslint.org/docs/rules/space-in-parens}
+*/
+rules[ 'space-in-parens' ] = [ 'error', 'always' ];
+
+/**
+* Enforces consistent spacing after the `//` or `/*` in a comment.
+*
+* @name spaced-comment
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'always', { 'markers': [ '/' ] }]
+* @see [spaced-comment]{@link https://eslint.org/docs/rules/spaced-comment}
+*/
+rules[ 'spaced-comment' ] = [ 'error', 'always', {
+	'markers': [ '/' ]
+}];
+
+/**
+* Enforces the use of `isNaN()` when checking for `NaN`.
+*
+* @name use-isnan
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [use-isnan]{@link https://eslint.org/docs/rules/use-isnan}
+*
+* @example
+* // Bad...
+* if (foo === NaN) {
+*   // ...
+* }
+*
+* // Good...
+* if (isNaN(foo)) {
+*   // ...
+* }
+*/
+rules[ 'use-isnan' ] = 'error';
+
+/**
+* Enforces the style of conditions to avoid Yoda conditions.
+*
+* @name yoda
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [yoda]{@link https://eslint.org/docs/rules/yoda}
+*
+* @example
+* // Bad...
+* if ('red' === color) {
+*   // ...
+* }
+*
+* // Good...
+* if (color === 'red') {
+*   // ...
+* }
+*/
+rules[ 'yoda' ] = 'error';
```

Comment:
Looks like we duplicate a number of rules from our normal ESLint configurations, with some specific overrides. Is it possible to avoid duplication? Perhaps we could consider a refactor in which individual configurations are exported as tiny standalone modules which can be independently imported. I think part of the issue is that rules are defined and set in much larger files. If we atomized, we could encourage a bit more reuse.

Reply:
@kgryte Yeah; we could look into that. We may also want to look which of our custom `stdlib` ESLint rules, if any, would make sense to be enabled for TypeScript files. I would prefer to get this PR merged before making such optimizations, though

Reply:
That sounds reasonable to me.

Reply:
I suggest creating a tracking issue for that which lists all current lint rules and indicates which may be applicable.

Reply:
@Planeshifter What is the current status with this? I am not sure if you have already determined applicability or whether this remains to be done.

Reply:
Created an internal TODO.